Add Authorization to alertmanager http configs (#4110)
prometheus-operator/prometheus-operator: Documentation/api.md

[Back to TOC](#table-of-contents)

## Authorization

> Authorization allow an endpoint to authenticate by configuring the authentication token More info: https://prometheus.io/docs/operating/configuration/#endpoints

Reviewer: ```suggestion
Authorization allows configuring authentication with endpoint via a token. More info: https://prometheus.io/docs/operating/configuration/#endpoints
```

| Field | Description | Scheme | Required |
| ----- | ----------- | ------ | -------- |
| authorization | Authorization header configuration for the client. This is mutually exclusive with BasicAuth | *monitoringv1.Authorization | false |

> | basicAuth | BasicAuth for the client. This is mutually exclusive with Authorization | *monitoringv1.BasicAuth | false |

Reviewer 2: Would it make sense to deprecate `basicAuth` field?
